src/internal/connector/onedrive/collection.go

```diff
@@ -258,9 +267,38 @@ func (oc *Collection) populateItems(ctx context.Context) {
 				metaSuffix = DirMetaFileSuffix
 			}
 
-			// Fetch data for files
+			// Fetch data for the file
+			for i := 1; i <= maxRetries; i++ {
+				itemInfo, itemData, err = oc.itemReader(ctx, item)
+
+				// retry on Timeout type errors, break otherwise.
+				if err == nil || graph.IsErrTimeout(err) == nil {
+					break
+				}
+
+				if i < maxRetries {
+					time.Sleep(1 * time.Second)
+				}
+			}
+
+			if err != nil {
+				errUpdater(*item.GetId(), err)
+				return
+			}
+
+			// Fetch metadata for the file
+			// TODO(meain): refactor these multiple retry logic
 			for i := 1; i <= maxRetries; i++ {
-				itemInfo, itemData, itemMeta, err = oc.itemReader(ctx, oc.service, oc.driveID, item)
+				if !oc.ctrl.ToggleFeatures.DisablePermissionsBackup {
+					itemMeta, itemMetaSize, err = oc.itemMetaReader(ctx, oc.service, oc.driveID, item)
+				} else {
+					// We are still writing the metadata file but with
+					// empty permissions as we are not sure how the
+					// restore will be called.
+					itemMeta = io.NopCloser(strings.NewReader("{}"))
+					itemMetaSize = 2
+					err = nil
+				}
```

Comment:
it's a little hard to follow when we set err = nil so we hit the break in the else-block. It would be more straightforward to read if we just make an if-block that breaks when needed and let everything else have it's own control flow
```suggestion
				if oc.ctrl.ToggleFeatures.DisablePermissionsBackup {
					// We are still writing the metadata file but with
					// empty permissions as we are not sure how the
					// restore will be called.
					itemMeta = io.NopCloser(strings.NewReader("{}"))
					itemMetaSize = 2

					break
				}

				itemMeta, itemMetaSize, err = oc.itemMetaReader(ctx, oc.service, oc.driveID, item)
```

Reply:
Woops, this is such an obvious refactor :D

Reply:
no worries, it can be easy to miss stuff like this if there's a lot of other things going on too 😄